packages/lesswrong/components/comments/CommentsItem/CommentUserName.tsx

```diff
@@ -51,9 +51,9 @@ const CommentUserName = ({comment, classes, simple = false, isPostAuthor, hideSp
   
   if (comment.deleted) {
     return <span className={className}>[comment deleted]</span>
-  } else if (comment.hideAuthor || !author) {
+  } else if (comment.hideAuthor || !author || author.deleted) {
     return <span className={className}>
-      <UserNameDeleted/>
+      <UserNameDeleted userShownToAdmins={author} />
```

Comment:
The code around displaying usernames is surprisingly maze-like. I added this bit to reach `UserNameDeleted` before we set up the user flair (i.e. the icons) below.